include/deal.II/fe/fe_interface_values.h

```diff
@@ -2242,35 +2412,83 @@ FEInterfaceValues<dim, spacedim>::reinit(
+          fe_face_values_neighbor = internal_fe_face_values_neighbor.get();
+        }
+      else
+        {
+          internal_fe_subface_values_neighbor->reinit(cell_neighbor,
+                                                      face_no_neighbor,
+                                                      sub_face_no_neighbor);
+          fe_face_values_neighbor = internal_fe_subface_values_neighbor.get();
+        }
+
+      AssertDimension(fe_face_values->n_quadrature_points,
+                      fe_face_values_neighbor->n_quadrature_points);
+
+      const_cast<unsigned int &>(this->n_quadrature_points) =
+        fe_face_values->n_quadrature_points;
     }
-  if (sub_face_no_neighbor == numbers::invalid_unsigned_int)
+  else if (internal_hp_fe_face_values)
     {
-      internal_fe_face_values_neighbor.reinit(cell_neighbor, face_no_neighbor);
-      fe_face_values_neighbor = &internal_fe_face_values_neighbor;
+      // Same as if above, but when hp is enabled.
+      if (sub_face_no == numbers::invalid_unsigned_int)
+        {
+          internal_hp_fe_face_values->reinit(cell, face_no);
+          fe_face_values = &const_cast<FEFaceValues<dim, spacedim> &>(
+            internal_hp_fe_face_values->get_present_fe_values());
+        }
+      else
+        {
+          internal_hp_fe_subface_values->reinit(cell, face_no, sub_face_no);
+          fe_face_values = &const_cast<FESubfaceValues<dim, spacedim> &>(
+            internal_hp_fe_subface_values->get_present_fe_values());
+        }
+      if (sub_face_no_neighbor == numbers::invalid_unsigned_int)
+        {
+          internal_hp_fe_face_values_neighbor->reinit(cell_neighbor,
+                                                      face_no_neighbor);
+
+          fe_face_values_neighbor = &const_cast<FEFaceValues<dim, spacedim> &>(
+            internal_hp_fe_face_values_neighbor->get_present_fe_values());
+        }
+      else
+        {
+          internal_hp_fe_subface_values_neighbor->reinit(cell_neighbor,
+                                                         face_no_neighbor,
+                                                         sub_face_no_neighbor);
+
+          fe_face_values_neighbor =
+            &const_cast<FESubfaceValues<dim, spacedim> &>(
+              internal_hp_fe_subface_values_neighbor->get_present_fe_values());
+        }
+
+      AssertDimension(fe_face_values->n_quadrature_points,
+                      fe_face_values_neighbor->n_quadrature_points);
+
+      const_cast<unsigned int &>(this->n_quadrature_points) =
+        fe_face_values->n_quadrature_points;
     }
   else
     {
-      internal_fe_subface_values_neighbor.reinit(cell_neighbor,
-                                                 face_no_neighbor,
-                                                 sub_face_no_neighbor);
-      fe_face_values_neighbor = &internal_fe_subface_values_neighbor;
+      Assert(false, ExcInternalError());
```

Comment:
If you formulate this check as
```
Assert(internal_fe_face_values || internal_hp_fe_face_values, ExcNotInitialized());
```
then it would render a more expansive error message when triggered.

Same elsewhere.